src/Refitter.Tests/Examples/UseSystemTextJsonPolymorphicSerializationTests.cs

```diff
@@ -0,0 +1,247 @@
+      properties:
+        $type:
+          type: string
+        typeId:
+          type: integer
+          format: int64
+      additionalProperties: false
+      discriminator:
+        propertyName: $type
+    SomeComponentState:
+      enum:
+        - Active
+        - Inactive
+        - Blocked
+        - Deleted
+      type: string
+    SomeComponentType:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/Component'
+      properties:
+        state:
+          $ref: '#/components/schemas/SomeComponentState'
+        isBaseRole:
+          type: boolean
+        name:
+          type: string
+          nullable: true
+        numberingId:
+          type: string
+          nullable: true
+      additionalProperties: false
+    Component:
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int64
+        metadata:
+          $ref: '#/components/schemas/Metadata'
+      additionalProperties: false
+    LoadingAddress:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/SomeComponent'
+      properties:
+        info:
+          type: string
+          nullable: true
+      additionalProperties: false
+    Warehouse:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/SomeComponent'
+      properties:
+        info:
+          type: string
+          nullable: true
+      additionalProperties: false
+    WarehouseResponse:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/SomeComponent'
+      properties:
+        info:
+          type: string
+          nullable: true
+      additionalProperties: false
+    UserComponent:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/SomeComponent'
+      properties:
+        info:
+          type: string
+          nullable: true
+      additionalProperties: false
+    UserComponent2:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/UserComponent'
+      properties:
+        info2:
+          type: string
+          nullable: true
+      additionalProperties: false
+    ProblemDetails:
+      required:
+        - $type
+      type: object
+      properties:
+        $type:
+          type: string
+        type:
+          type: string
+          nullable: true
+        title:
+          type: string
+          nullable: true
+        status:
+          type: integer
+          format: int32
+          nullable: true
+        detail:
+          type: string
+          nullable: true
+        instance:
+          type: string
+          nullable: true
+      additionalProperties: { }
+      discriminator:
+        propertyName: $type
+";
+
+    [Fact]
+    public async Task Can_Generate_Code()
+    {
+        string generateCode = await GenerateCode();
+        generateCode.Should().NotBeNullOrWhiteSpace();
+    }
+
+    [Fact]
+    public async Task Use_SystemTextJson_Polymorphic_Serialization()
```

Comment:
@ryanheath This test seems to fail. Can you please fix it?

Reply:
It only seems to fail when run in parallel, I'm not sure why.

Reply:
@ryanheath I pulled your branch and tried it out locally. You're right, it only fails when run in parallel. The tests are correct and the structure is fine. I can't immediately see what could be wrong but the generated code is quite different...

When run individually, it generates 244 lines of code, when run in parallel it generates 375 lines of code 🤯